poll/poll.py

```diff
@@ -823,7 +823,7 @@ def generate_report_data(self, user_state_iterator, limit_responses=None):
                 continue
             report = {
                 self.ugettext('Question'): self.question,
-                self.ugettext('Answer'): answers_dict[choice]['label'],
+                self.ugettext('Answer'): answers_dict.get(choice, {}).get('label', ''),
```

Comment:
Rather than return an empty string, I think in this case it should return choice id. 
```suggestion
                self.ugettext('Answer'): answers_dict.get(choice, {}).get('label') or choice,
```

Reply:
The choice id in place of answer in the report may confuse the user. This may be logged as another bug in the future.

Reply:
Which user will it confuse? This is data available to instructors only. 
If a simple id seems confusing, you can format the choice to be cleaner, so something like: "[Removed Poll Option {choice}]" 
Or something with similar language. 

In any cases an empty response is also possible if a student hasn't responded, so leaving the option empty is bound to create more confusion. It is also not accurate because the user has made an choice, and issue is in resolving that choice to a label, not in getting the choice in the first place.

I will also note that this bug also exists for surveys, and you may want to fix that while you're at it.